request.go

```diff
@@ -324,9 +366,23 @@ func unmarshalNode(data *Node, model reflect.Value, included *map[string]*Node)
 				}
 
 				fieldValue.Set(m)
+			}
+
+		} else if annotation == annotationLinks {
```

Comment:
This is to handle when the a Resource struct has a `jsonapi:"links"` tag directly on it. In which case, we just unmarshal it directly.